### ISSUE TITLE
Can this frontend be integrated with cockpit?

### ISSUE THREAD

**Issue by emaayan** [cockpit-project/cockpit#20761]
So i recently just saw this
https://github.com/xinthose/Netplan-GUI

I haven't played with it yet but i was wondering if somehow a "bridge" in a form of a cockpit plugin could be created so it could be used with cockpit

These days everytime i use a new ubuntu server lts (now 24.04 lts ) i have to "get creative" and find automated ways to rip out the networkd layer and use networkmanager during cloud init install just so that cockput could manage the network..

**Comment by martinpitt** [comment]
I was hoping that https://ubuntu.com/blog/netplan-configuration-across-desktop-server-cloud-and-iot would help with making the Networking page more compatible with networkd..

**Comment by emaayan** [comment]
> I was hoping that [ubuntu.com/blog/netplan-configuration-across-desktop-server-cloud-and-iot](https://ubuntu.com/blog/netplan-configuration-across-desktop-server-cloud-and-iot) would help with making the Networking page more compatible with networkd..

how? were you planning on using [libnetplan ](https://netplan.readthedocs.io/en/0.106.1/netplan-dbus/#netplan-dbus)? i've suspected they made some changes to NetworkManager cause suddenly i have no default profile in network manager when i switch it on.

**Comment by martinpitt** [comment]
No, I thought the idea of that Ubuntu feature was that changing things through NM's API would write out actual netplan files. I never tried it, it was just a remark.

It's safe to say that nobody in the cockpit team will work on this anytime soon, we are swamped with other stuff, and Red Hat has no interest in netplan. So if anyone wants to have a go at this, please do -- but please talk to us *first* before creating a huge PR :wink:
